src/include/txn_inline.h

```diff
@@ -451,8 +451,8 @@ __wt_txn_oldest_id(WT_SESSION_IMPL *session)
      * Take a local copy of these IDs in case they are updated while we are checking visibility.
      */
     oldest_id = txn_global->oldest_id;
-    include_checkpoint_txn = btree == NULL ||
-      (!WT_IS_HS(btree) && btree->checkpoint_gen != __wt_gen(session, WT_GEN_CHECKPOINT));
+    include_checkpoint_txn =
```

Comment:
I believe this change is causing some `gen_drain` issues. But ignoring the checkpoint transaction id for history store can lead to wrong global visibility during a checkpoint is in progress.